pages/courses/[id]/[section]/[lesson].js

```diff
@@ -58,47 +58,70 @@ function Lessons({ course, lesson, currentDate }) {
     }
   }, [cohorts, user])
 
-  useEffect(async () => {
-    setLessonsSubmitted(await getLessonsSubmissions(user?.uid))
-  }, [user, open])
+  useEffect(() => {
+    const fetchLessonsSubmitted = async () => {
+      let lessonsSubmitted_ = await getLessonsSubmissions(user?.uid, cohort?.id)
+      setLessonsSubmitted(lessonsSubmitted_)
+    }
+    fetchLessonsSubmitted()
+  }, [user, cohort, open])
 
   useEffect(() => {
-    lessonsSubmitted.map((item) => {
-      if (item?.lesson === lesson && item?.user == user?.uid && item?.cohort_id == cohort?.id) {
+    lessonsSubmitted.forEach((item) => {
+      console.log('lesson', lesson)
+      if (item?.lesson === lesson) {
         setUserSubmission(item.content.value)
-        validateUserSubmission(item.content.value)
         setLessonSent(true)
       }
     })
-  }, [lessonsSubmitted])
+  }, [lessonsSubmitted, lesson])
 
   useEffect(() => {
     setSortedLessons(course.lessons.sort((a, b) => (a.section > b.section ? 1 : -1)))
   }, [sortedLessons])
+
+  useEffect(() => {
```

Comment:
This code should not be here. It is a generic language change code that should be in the page template or navbar component.

Reply:
Removed 😉